summarize_pr.py

```diff
@@ -42,29 +40,113 @@ def summarize_pr(
-        "Summary:",
+        post_prompt,
     ]
 
     print("---Prompt---\n", prompt)
     response = model.generate_content(prompt)
     print("---Gemini Response---\n", response)
+    return response.text
+
+
+def analyze_pr(api: GhApi, pr_number: str, model_id: str) -> Tuple:
+    pull_request_content = _get_pr_content(api, pr_number)
+
+    summarize_output = _process_code_base(
+        pull_request_content,
+        model_id,
+        "Your task is to summarize the Pull Request in Markdown format.",
+        post_prompt="Summary:",
+    )
+
+    analysis_output = _process_code_base(
+        pull_request_content,
+        model_id,
+        task_prompt="Your task is to analyze each code file line by line, then output any bugs you find and improvements that can be made to the code quality for readability and maintainability.",
+        post_prompt="Analysis:",
+    )
+
+    latest_commit = api.pulls.list_commits(pr_number)[-1]["sha"]
+
+    # Create a comment on the PR
+    comment_content = f"""# Pull Request Summary from Gemini ✨\n{summarize_output}\n\n# Code Analysis from Gemini ✨\n{analysis_output}\n Generated at Commit: `{latest_commit}`"""
 
     # Post the summary as a comment
-    api.issues.create_comment(
-        pr_number, f"## Pull Request Summary from Gemini\n {response.text}"
+    api.issues.create_comment(pr_number, comment_content)
+
+    return summarize_output, analysis_output
+
+
+def convert_to_sarif(model_id: str, analysis_content: str) -> str:
+    model = GenerativeModel(
+        model_id,
+        system_instruction=[
+            "You are an expert software engineer.",
+        ],
     )
 
+    generation_config = GenerationConfig(
+        response_mime_type="application/json",
+        max_output_tokens=8192,
+    )
+
+    sarif_schema = requests.get(
+        "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/123e95847b13fbdd4cbe2120fa5e33355d4a042b/Schemata/sarif-schema-2.1.0.json"
+    ).text
+
+    prompt = [
+        "The following is analysis of a GitHub Pull Request.",
+        "Your task is to convert this into SARIF (Static Analysis Results Interchange Format) format.",
+        "This is the SARIF schema:",
+        sarif_schema,
+        "Analysis Content:",
+        analysis_content,
+        "SARIF:"
+        """{"$schema": "https://json-schema.org/draft/2020-12/schema#", "version": "2.1.0", "runs": [{"tool": {"driver": {"name": "Gemini", "shortDescription": {"text": "A large language model capable of generating text, translating languages, writing different kinds of creative content, and answering your questions in an informative way."}, "fullDescription": {"text": "Gemini Pro is a large language model (LLM) developed by Google AI. It is trained on a massive dataset of text and code, and can be used for a wide variety of tasks, including text generation, translation, writing different kinds of creative content, and answering your questions in an informative way."}}}, "results":""",
```

Comment:
The inclusion of a SARIF generation capability adds a valuable feature to the code. The usage of the Gemini model for SARIF conversion is well-implemented. 

Consider adding a comment explaining the purpose of the `convert_to_sarif` function and the rationale for using the specific SARIF schema. Additionally, it would be beneficial to explore the possibility of adding validation steps to ensure that the generated SARIF output adheres to the specified schema.